lib/spack/spack/mirror.py

```diff
@@ -154,8 +154,68 @@ def push_url(self):
         """Get the valid, canonicalized fetch URL"""
         return self.get_url("push")
 
+    def validate(self, direction="push"):
```

Comment:
Probably better is `ensure_credentials_set` or something. Any function that returns None and raises is called ensure_* elsewhere.

Also add type hints.

And drop the comment about "URL is accessible". The only way of knowing for a fact is successfully pushing...

Reply:
Sure, I originally had it return bool but decided to raise instead and reuse the `MirrorError`.l

For `s3`, and I think also`oci`, mirrors you can make a request to stat if the bucket/prefix exists.

Reply:
We could also do a test write here on push and a test read on fetch..I'll drop the comment but put it on my backlog